### ISSUE TITLE
TypeLoadException due to BLOCK_REGION_UNION

### ISSUE THREAD

**Issue by alexrp** [dahall/Vanara#55]
**Describe the bug**
```
System.TypeLoadException
  HResult=0x80131522
  Message=Could not load type 'BLOCK_REGION_UNION' from assembly 'Vanara.PInvoke.Kernel32, Version=2.3.10.0, Culture=neutral, PublicKeyToken=c37e4080322237fa' because it contains an object field at offset 0 that is incorrectly aligned or overlapped by a non-object field.
```

**What code is involved**
No code, simply referencing several of the Vanara assemblies seems to trigger it. Not exactly sure which, but my project is referencing these:

* Vanara.PInvoke.Mpr
* Vanara.PInvoke.NtDll
* Vanara.PInvoke.Shell32
* Vanara.PInvoke.User32
* Vanara.PInvoke.User32.Gdi
* Vanara.PInvoke.WinINet

(And all dependencies of these.)

**Expected behavior**
No TLE.

**Comment by alexrp** [comment]
Hm. The reason it shows up is probably because I'm using MEF in this project, which scans over all referenced assemblies and their types, forcing them to be loaded.

**Comment by alexrp** [comment]
Happens both with 2.3.9 and 2.3.10.

**Comment by dahall** [comment]
I applied and posted a fix just now that should resolve this issue. If you can build, please test. Otherwise, I'll include in my next release.

**Comment by alexrp** [comment]
I think a simple test case would be something along these lines:

```csharp
using System;
using System.IO;
using System.Reflection;
using System.Runtime.CompilerServices;

class Program
{
    static void Main()
    {
        foreach (var file in Directory.EnumerateFiles(".", "Vanara*.dll"))
            foreach (var type in Assembly.LoadFrom(file).DefinedTypes)
                RuntimeHelpers.RunClassConstructor(type.TypeHandle);
    }
}
```

Place all the Vanara assemblies next to this program and it'll load all of them and force all types in them to be initialized, which will result in exception(s) if any of them have invalid layout.

**Comment by dahall** [comment]
Fixed and released today in 2.3.11
